Content.Server/Chat/Systems/ChatSystem.cs

```diff
@@ -734,14 +822,56 @@ private MessageRangeCheckResult MessageRangeCheck(ICommonSession session, ICChat
     /// <summary>
     ///     Sends a chat message to the given players in range of the source entity.
     /// </summary>
-    private void SendInVoiceRange(ChatChannel channel, string message, string wrappedMessage, EntityUid source, ChatTransmitRange range, NetUserId? author = null)
+    private void SendInVoiceRange(
+        ChatChannel channel,
+        string name,
+        string message,
+        string wrappedMessage,
+        string obfuscated,
+        string obfuscatedWrappedMessage,
+        EntityUid source,
+        ChatTransmitRange range,
+        NetUserId? author = null,
+        bool checkLOS = false
+        )
     {
+        // Get the current pressure at the point of the voice source.
+        var currentSourcePressure = 0f;
+        var sourceTileMixture = _atmosphere.GetContainingMixture(source, true);
+        if (sourceTileMixture != null)
+            currentSourcePressure = sourceTileMixture.Pressure;
+
+        var minPresure = _configurationManager.GetCVar(L5CCVars.MinSoundTransmitPressure);
+        var inSpaceRange = _configurationManager.GetCVar(L5CCVars.InSpaceRange);
+
         foreach (var (session, data) in GetRecipients(source, VoiceRange))
         {
+            if (channel != ChatChannel.LOOC && channel != ChatChannel.Emotes && session.AttachedEntity != null)
+            {
+                // Get the current pressure at the point of the recipient
+                var currentRecipientPressure = 0f;
+                var recipientTileMixture = _atmosphere.GetContainingMixture(session.AttachedEntity.Value, true);
+                if (recipientTileMixture != null)
+                    currentRecipientPressure = recipientTileMixture.Pressure;
+
+                // If the pressure at the source or recipient is too low, set the transmit range to ~touch distance.
+                float transmitRange = VoiceRange;
+                if (currentSourcePressure < minPresure || currentRecipientPressure < minPresure)
+                    transmitRange = inSpaceRange;
+
+                if (session.AttachedEntity != null && Transform(source).Coordinates.TryDistance(EntityManager, Transform(session.AttachedEntity.Value).Coordinates, out var distance) && distance > transmitRange)
```

Comment:
Can get rid of first check: 
```suggestion
                if (Transform(source).Coordinates.TryDistance(EntityManager, Transform(session.AttachedEntity.Value).Coordinates, out var distance) && distance > transmitRange)
```